pnpm/src/main.ts

```diff
@@ -200,7 +200,7 @@ export async function main (inputArgv: string[]): Promise<void> {
     const filterResults = await filterPackagesFromDir(wsDir, filters, {
       engineStrict: config.engineStrict,
       nodeVersion: config.nodeVersion ?? config.useNodeVersion,
-      patterns: config.workspacePackagePatterns,
+      patterns: config.workspacePackagePatterns ?? ['.', '**'],
```

Comment:
Here's my prior comments on this here:

- https://github.com/pnpm/pnpm/pull/8215#discussion_r1641967881
- https://github.com/pnpm/pnpm/pull/8215#discussion_r1642037025

I think the `?? ['.', '**']` is ugly and should be `[]` instead, but this would be a behavior change and breaks a few tests. I'm looking into fixing these tests, but wanted to push up a rebase for this PR in the meantime.

Reply:
I have made patterns optional, not default value is needed.

Reply:
One thing I was worried about with `patterns` being optional is developers in the future writing code in pnpm forgetting to pass in a value when calling `findWorkspacePackages`. Everything might work as expected when testing, but in production packages might get included from the default filter `['.', '**']` that shouldn't, and then users will get confused.

I think that's a general footgun with optional parameters.